### ISSUE TITLE
Google Anruferbilder

### ISSUE THREAD

**Issue by bahnuhr** [soef/ioBroker.tr-064#52]
Guten Tag,

das holen der Anruferbilder klappt immer noch nicht (bzw. ich hab es noch nicht hinbekommen).

Genauer:
Wir hatten für deine Version 0.3.6 und 0.3.7 mal diskuttiert wie man diese Bilder mit phonebook holt.
Es wird auch eine url angezeigt; diese funktioniert jedoch nicht.

Beispiel:
Ich gebe in iobroker bei phonebook.number und sofort wird mit bei phonebook.image eine url angezeigt.
So ungefähr (Zahlen teilweise geändert):
http://192.168.xxx.x:49000/download.lua?path=https://www.google.com/m8/feeds/photos/media/d.muxxxxx%40googlemail.com/7f3c56c1xxxxxx,240&sid=b40d379a18f1xxxxxx=240

Wenn ich diese url in chrome auf win7 eingebe kommt:
`<s:Envelope xmlns:s="http://schemas.xmlsoap.org/soap/envelope/" s:encodingStyle="http://schemas.xmlsoap.org/soap/encoding/">
<s:Body>
<s:Fault>
<faultcode>s:Client</faultcode>
<faultstring>UPnPError</faultstring>
<detail>
<UPnPError xmlns="urn:dslforum-org:control-1-0">
<errorCode>606</errorCode>
<errorDescription>Action Not Authorized</errorDescription>
</UPnPError>
</detail>
</s:Fault>
</s:Body>
</s:Envelope>`

Fazit:
Irgendwie stimmt da etwas nicht. Ich habe es bisher nicht hinbekommen, dass mir die Bilder von den Anrufern angezeigt werden.

Gibt es hier schon eine Lösung ?

mfg
D.Müller

**Comment by soef** [comment]
Hallo, der Fehler hier ist eine ungültige Session-ID in der URL. Aber auch mit einer gültigen Session-ID funktioniert es mit dem Google-Adressbuch nicht. In der Fritz Fon App von AVM wird das Bild auch nicht angezeigt...

**Comment by bahnuhr** [comment]
Hallo,
im Telefon Fritz C5 funktionieren die Bilder. Also kommen die doch irgendwie richtig in der Fritzbox an.
Kannst du da etwas machen?
Ich vermute es hängt an der Google Auth Identifizierung.

mfg
Dieter

**Comment by soef** [comment]
Hallo, ja, das wird wohl an der Google Authentisierung liegen. Die Fritzbox kennt den API-Key und das Token und wird dieses wahrscheinlich an das Fritz C5 weitergeben, so das es damit auch authentisieren kann. Das einzige was bleibt um die Grafiken von Google zu bekommen, ist sich zuvor auch bei Google anzumelden.  Das muss aber dort gemacht werden auf die Grafik zugegriffen und sie angezeigt werden soll.

**Comment by bahnuhr** [comment]
Hallo soef,
habe auf dem Handy die app "boxtogo". Diese greift ja auch auf die fritz.box zu.
Und hier werden auch die Bilder angezeigt.
Nur zur Info.
Vielleicht besteht ja noch eine Möglichkeit, dass dies auch mit dem Adapter funktioniert.
mfg
Dieter

**Comment by soef** [comment]
Hallo, die App boxtogo verwendet den Push Service der FritzBox, also nicht das TR064Protokoll. Ich habe deren Daten bei mir nicht eingetragen, da ich den Dienst mit eigenen Daten nutze. Wenn ich es richtig sehe, ist aber das Bild bei der versendeten Push-EMail nicht enthalten. Vielleicht holt die App sich das Bild aus dem lokalen Adressbuch des Handys.

**Comment by bahnuhr** [comment]
Schade, schade, dass wir dies nicht gelöst bekommen.

**Comment by soef** [comment]
Du kannst ja einmal eine Mail an AVM schreiben und fragen, ob da im Google Telefonbuch bezüglich der Bilder ein Fehler vorliegt. Vielleicht mit dem Hinweis dass das Bild auch in der Fritz Fon App fehlt.

**Comment by bahnuhr** [comment]
Ich hatte schon mal Mailkontakt mit AVM. Das bringt nichts.
Die Personen die mir geschrieben hatten, wollten oder konnten mich nicht verstehen.
mfg
P.S.
Wenn wir dies hier nicht gelöst bekommen, dann klappt das eben nicht.

**Comment by soef** [comment]
Frag doch AVM einfach einmal warum das Bild in der Fritz Fon App nicht angezeigt wird. Das kannst du ja auch im App, bzw. Play Store bei den Rezensionen posten. Darauf müsste AVM reagieren.

**Comment by soef** [comment]
Hallo, die Antwort von AVM passt ja nicht zu der Frage, Warum wird das Bild in der Fritz Fon (Smartphone) App nicht angezeigt. Ein C5 habe ich leider nicht.

**Comment by bahnuhr** [comment]
Hallo soef,
ich klink mich hier jetzt aus.
Wir sind ja nun schon seit mehreren Monaten an diesem Thema dran.
Ich habe einfach keine Idee mehr.
Wenn von dir keine Lösung kommt, dann muss ich hier jetzt fachlich aufgeben. Ich bekomme dies ohne Hilfe nicht hin.
Und AVM will/kann da nicht helfen.

mfg
Dieter

**Comment by jogibear9988** [comment]
This issue was moved to iobroker-community-adapters/ioBroker.tr-064-community#10

**Comment by bahnuhr** [comment]
Der Pfad zum image ist ja bekannt und wird in das Objekt:
tr-064-community.0.callmonitor.connect.imageurlcaller
geschrieben.

Wäre es nicht möglich, dass das Bild gleich in einen Netzwerkordner als jpg geschrieben wird.
Man würde damit die zeitlich befristete sid der fritzbox umgehen.

**Comment by jogibear9988** [comment]
das ist doch sehr individuell. das kann doch mit dem javascript adapter gelöst werden.

**Comment by bahnuhr** [comment]
Das sehe ich nicht so. Wenn man schon einen Pfad reinschreibt dann sollte dieser auch funktionieren. Und das ist nicht so.
Denn der Pfad ist durch die temporäre sid nicht mehr verfügbar.
Deshalb Bild speichern und gut ist.

**Comment by jogibear9988** [comment]
Ja, hast vlt. schon recht. Im Endeffekt sollten die URLs funktionieren, zumindest wenn der Event passiert. Aber das speichern auf dem Netzwerk, der nächste möchte es auf Google Drive oder via FTP. Das sehe ich schon in extra adaptern die der User sich dann entsprechend verknüpft

**Comment by bahnuhr** [comment]
Nein.
die sid ist nach ein paar Sekunden nicht mehr gültig.
Also hilft der pfad nichts.
Helfen würde aber das gespeicherte Bild.
